yalu102/jailbreak.m

```diff
@@ -710,6 +710,22 @@ void exploit(void* btn, mach_port_t pt, uint64_t kernbase, uint64_t allprocs)
         
         WriteAnywhere32(v_mount + 0x71, v_flag);
     }
+    
+        { //Prevents OTA Daemons from launching
+        
+        int f1 = open("/System/Library/LaunchDaemons/com.apple.mobile.softwareupdated.bak", O_RDONLY);
+        int f2 = open("/System/Library/LaunchDaemons/com.apple.softwareupdateservicesd.bak", O_RDONLY);
+        
+        if ( (f1 == -1) || (f2 == -1) )  {
+            NSLog(@"Killing OTA Updates...");
+            system("mv -f /System/Library/LaunchDaemons/com.apple.mobile.softwareupdated.plist /System/Library/LaunchDaemons/com.apple.mobile.softwareupdated.bak");
+            system("mv -f /System/Library/LaunchDaemons/com.apple.softwareupdateservicesd.plist /System/Library/LaunchDaemons/com.apple.softwareupdateservicesd.bak");
+            NSMutableDictionary* prefDict = [[NSMutableDictionary alloc] initWithContentsOfFile:@"/var/mobile/Library/Preferences/com.apple.Preferences.plist"];
+            [prefDict setObject:[NSNumber numberWithBool:NO] forKey:@"kBadgedForSoftwareUpdateKey"];
+            [prefDict setObject:[NSNumber numberWithBool:NO] forKey:@"kBadgedForSoftwareUpdateJumpOnceKey"];
+            [prefDict writeToFile:@"/var/mobile/Library/Preferences/com.apple.Preferences.plist" atomically:YES];
```

Comment:
don't have an iOS 10 device here to check the permissions, but it wouldn't hurt to make sure this file is owned by mobile after you modify it.